k8s/models/cronjob.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+
+# Copyright 2017-2019 The FIAAS Authors
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#      http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+
+import six
+
+from .common import ObjectMeta, ObjectReference, Time, ListMeta
+from .job import JobTemplateSpec
+from ..base import Model
+from ..fields import Field, ListField
+
+
+class CronJobSpec(Model):
+    concurrencyPolicy = Field(six.text_type) 
+    failedJobsHistoryLimit = Field(int)
+    jobTemplate = Field(JobTemplateSpec)
+    schedule = Field(six.text_type)
+    startingDeadlineSeconds = Field(int)
+    successfulJobsHistoryLimit = Field(int)
+    suspend = Field(bool)
+
+
+class CronJobStatus(Model):
+    active = ListField(ObjectReference)
+    lastScheduleTime = Field(Time)
+
+
+class CronJob(Model):
+    class Meta:
+        list_url = "/apis/batch/v1beta1/cronjobs"
+        url_template = "/apis/batch/v1beta1/namespaces/{namespace}/cronjobs/{name}"
+
+    metadata = Field(ObjectMeta)
+    spec = Field(CronJobSpec)
+    status = Field(CronJobStatus) 
+
+
+class CronJobList(Model):
```

Comment:
We have typically not defined the various `List` models, because they are mostly an artifact of the API. When working with collections of objects, the API will use the corresponding `List` object, but in the library it will simply be a Python list or generator of the main type. So you will never work directly with a `CronJobList` object, you will just work with regular Python lists of `CronJob` objects. You can skip this and any other `List` models.

Reply:
I've noticed it in the other models, however when I try to access it, I couldn't get working without the CronJobList.
I am not sure I understand how does it works, but will remove it, if this is not necessary.

Reply:
The way you would get a list of CronJobs is to call `CronJob.list` or `CronJob.find`. These methods will call API-methods that returns `CronJobList` objects, but the client will deal with that and just return a plain Python list of `CronJob` objects.